File: backend/app/models/vehiculo.py
from sqlalchemy import Column, ForeignKey, Integer, String
from app.database import Base

from sqlalchemy.orm import relationship



class Vehiculo(Base):
    __tablename__ = "vehiculos"

    id = Column(Integer, primary_key=True, index=True)
    marca  = Column(String,  nullable=False)
    modelo = Column(String,  nullable=False)
    anio   = Column(Integer, nullable=False)
    placa  = Column(String,  nullable=False)
    kilometraje_actual = Column(Integer, nullable=False, default=0)
    usuario_id = Column(Integer, ForeignKey("usuarios.id"), unique=True, nullable=False)

    mantenimientos = relationship(
        "Mantenimiento",
        back_populates="vehiculo",
        cascade="all, delete-orphan",
    )
    intervalos = relationship(
        "IntervaloVehiculo",
        back_populates="vehiculo",
        cascade="all, delete-orphan",
    )
    usuario = relationship("Usuario", back_populates="vehiculo")

File: mobile/lib/models/vehiculo.dart
class Vehiculo {
  const Vehiculo({
    required this.id,
    required this.marca,
    required this.modelo,
    required this.anio,
    required this.placa,
    required this.kilometrajeActual,
  });

  final int id;
  final String marca;
  final String modelo;
  final int anio;
  final String? placa;
  final int kilometrajeActual;

  factory Vehiculo.fromJson(Map<String, dynamic> json) {
    return Vehiculo(
      id: json['id'] as int,
      marca: json['marca'] as String,
      modelo: json['modelo'] as String,
      anio: json['anio'] as int,
      placa: json['placa'] as String?,
      kilometrajeActual: json['kilometraje_actual'] as int,
    );
  }
}

class DatosNuevoVehiculo {
  const DatosNuevoVehiculo({
    required this.marca,
    required this.modelo,
    required this.anio,
    required this.placa,
  });

  final String marca;
  final String modelo;
  final int anio;
  final String placa;

  Map<String, dynamic> toJson() {
    return {'marca': marca, 'modelo': modelo, 'anio': anio, 'placa': placa};
  }
}
